s3/s3_operations.cpp

```diff
@@ -2231,6 +2266,16 @@ namespace irods_s3 {
 
     irods::error s3_notify_operation( irods::plugin_context& _ctx,
         const std::string* str ) {
+        if (is_cacheless_mode(_ctx.prop_map())) {
+            // Go ahead and update the physical_path before a potential
+            // redirect so the DB gets updated properly but only do so if
+            // we are creating a new object. Otherwise, just use the path
+            // that already existed as what is in the DB must be honored.
+            irods::file_object_ptr object = boost::dynamic_pointer_cast<irods::file_object>(_ctx.fco());
+            if (object->flags() & O_CREAT) {
+                update_physical_path_for_decoupled_naming(_ctx);
```

Comment:
Does the O_CREAT flag get removed if it is used as part of, e.g. an overwrite via `iput -f`? Or is that API dependent? In other words, Is the check here sufficient to determine whether a new object is being created? Or maybe `update_physical_path_for_decoupled_naming` figures it out?

Reply:
Well, this didn't even work as I must have made a mistake in my manual testing.

Instead of using the open flags (which turned out to be zero), I am now checking the openType is CREATE_TYPE.  That seems to be working fine.  I verified the force overwrite, the openType is not CREATE_TYPE thus the physical_path is not updated meaning whatever was used before remains.